src/v/archival/ntp_archiver_service.cc

```diff
@@ -546,78 +546,14 @@ ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
           m.get_last_offset(),
           m.get_last_uploaded_compacted_offset());
 
-        if (m.get_last_offset() < manifest().get_last_offset()) {
-            // This indicates time travel, possible because the source cluster
-            // had a stale leader node upload a manifest on top of a more
```

Comment:
We should probably gate this on a feature flag, or don't sync if there's an on-going upgrade

Reply:
Great catch, added a feature flag check.